### ISSUE TITLE
How is skel/default/bin/dhcpcd.sh meant to work?

### ISSUE THREAD

**Issue by jsmeix** [rear/rear#3295]
Triggered by
https://github.com/rear/rear/issues/3285#issuecomment-2239071913
I tried to make sense how
skel/default/bin/dhcpcd.sh
is meant to work but I fail to understand it.

As far as I see we at SUSE do no longer provide dhcpcd
at least not in openSUSE Leap but likely since earlier
so I cannot try out dhcpcd with reasonable effort.

In
prep/GNU/Linux/210_include_dhclient.sh
I noticed
```
# We made our own /etc/dhclient.conf and /bin/dhclient-script files (no need to copy these
# from the local Linux system for dhclient). For dhcpcd we have /bin/dhcpcd.sh foreseen.
```

In
skel/default/etc/scripts/system-setup.d/58-start-dhclient.sh
there is
```
        case "$DHCLIENT_BIN" in
            ...
                (dhcpcd*)
                        "$DHCLIENT_BIN" ${DEVICE}
                    ;;
```
but I think this cannot work
because from plain looking at the code in
skel/default/bin/dhcpcd.sh
```
if [ $# -lt 2 ]; then
  logger -s -p local0.err -t dhcpcd.sh "wrong usage"
  exit 1
fi
```
it exits when there are less than 2 arguments.

**Comment by jsmeix** [comment]
@rear/contributors 
could you please (as time permits) have a look here?

**Comment by schlomo** [comment]
I'm wondering which distros actually use dhcpcd by default instead of dhclient?

**Comment by jsmeix** [comment]
@pcahyna @lzaoral 
do you know if Red Hat or Fedora based distros
may use dhcpcd instead of dhclient?

**Comment by jsmeix** [comment]
Regarding SUSE and openSUSE:

In openSUSE Factory (i.e. the "head of development"
wherefrom openSUSE Tubleweed gets built)
there is since 11. July 2024 a new package 'dhcpcd'
see
https://build.opensuse.org/package/show/openSUSE:Factory/dhcpcd
and
https://build.opensuse.org/request/show/1185096
where the latter tells about a reason behind:
```
New dhcp client package to allow replacing of the
deprecated dhclient from dhcp-server in e.g. cloud-init.
```

So in future SLES versions there will be likely 'dhcpcd'
but that is not something I worry now about.

When 'dhcpcd' will be in a future SLES version
I will have a look how that actually behaves - but
not earlier because currently I have other things to do
which have higher importance / priority than that.

**Comment by lzaoral** [comment]
> do you know if Red Hat or Fedora based distros may use dhcpcd instead of dhclient?

@jsmeix Yes, they even have to.  Our `rear` packages already use it on Fedora and RHEL 10 because `dhclient` is removed in RHEL 10 and deprecated in Fedora: https://fedoraproject.org/wiki/Changes/dhclient_deprecation

**Comment by jsmeix** [comment]
@lzaoral 
thank you for your prompt reply!

I think hereby you "won the price" to have a look
how skel/default/bin/dhcpcd.sh is meant to work ;-)

Would you mind when I assign this issue to you?

**Comment by lzaoral** [comment]
Sure!  Me or @pcahyna can take a look.  But beware that I have zero knowledge how `dhcpcd` work.  I have just swapped the `Requires:` in the spec file and it worked automagically with rear out of the box. 😄

**Comment by jsmeix** [comment]
@lzaoral 
my current only problem for the ReaR 3.0 release is
to find out if in skel/default/bin/dhcpcd.sh
sourcing the HostInfoFilePath e.g /var/lib/dhcpcd/dhcpcd-eth0.info
is sufficiently safe - in partcular if that may
(by chance, by accident) source third party code?
